### PR TITLE
Updates Service Broker dependencies

### LINKED ISSUE #206: Service broker dependencies are updated to latest versions
AC:
- [ ] Service broker dependencies (including Ruby / Rails versions) are updated to latest versions.
- [ ] NOTICES.txt is updated

### PR DESCRIPTION
**Pull request by diverdane**
### What does this PR do?
This change updates Service Broker dependencies. Some details:

- The Ruby version remains at 2.5
- The following gems have been updated via 'bundle install <gem>'

  ```
  conjur-api
  aruba
  pry-byebug
  license_finder
  spring
  activesupport
  railties
  actionview
  rack
  listen
  byebug
  cucumber
  json_spec
  rspec
  ci_reporter_rspec
  rspec_junit_formatter
  rest-client
  rspec-rails
  spring-watcher-listen
  ```

- The following gems were NOT updated, since updating them resulted in
  Service Broker functionality breakages:

  ```
  json-schema
  puma
  ```
  Additionally, the version of `puma` is fixed for now at version `3.12.5`
  because version `3.12.6` causes breakages.

- The `bundler-audit` gem was added so that we can run `bundle audit`
  to check for vulnerabilities in gems.
- The version of Postgres use in testing is updated to version 10.15
  (requiring the use of a database password).
- Notes on how to perform dependency updating was added to CONTRIBUTING.md.

### What ticket does this PR close?
Resolves #206 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -128,6 +128,32 @@ cd ci
...
+1. The dependency update that is done via `bundle install` may be too
+   aggressive in the sense that it could introduce new dependencies with
+   which the Conjur Service Broker is incompatible. In this case, you may
+   need to examine the errors in the Cucumber test output, or in the
+   Conjur Service Broker container logs to determine which dependency version
+   needs to be dialed back. You may need to manually adjust dependency
+   versions in the `Gemfile.lock` file.
+
+1. After tweaking the `Gemfile.lock` file to get the integration tests to
+   pass, command and push the updated dependencies, and create a pull
```
Comment (izgeri) [inline CONTRIBUTING.md:154]:
```suggestion
   pass, commit and push the updated dependencies, and create a pull
```
I assume?

File: docker-compose.yml [hunk 2]
```diff
@@ -2,13 +2,15 @@ version: "2"
 
 services:
   pg:
-    image: postgres:9.3
+    image: postgres:10.14
```
Comment (izgeri) [inline docker-compose.yml:5]: ah, good catch! should we go all the way to 10.15 since that's what we're using elsewhere?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -128,6 +128,32 @@ cd ci
...
+
+1. The dependency update that is done via `bundle install` may be too
+   aggressive in the sense that it could introduce new dependencies with
+   which the Conjur Service Broker is incompatible. In this case, you may
+   need to examine the errors in the Cucumber test output, or in the
+   Conjur Service Broker container logs to determine which dependency version
+   needs to be dialed back. You may need to manually adjust dependency
+   versions in the `Gemfile.lock` file.
+
+1. After tweaking the `Gemfile.lock` file to get the integration tests to
```
Comment (BradleyBoutcher) [inline CONTRIBUTING.md:153]: We should really avoid manually tweaking the `Gemfile.lock`. The best way to update dependencies is to use `bundle upgrade`. If we can't get that to work properly, then we can manually erase a line, then run `bundle upgrade` to fill it back in. Worst case scenario, just delete the `Gemfile.lock`, and run the tests, which will generate a new one.

Comment (diverdane) [reply CONTRIBUTING.md:169]: Yes, indeed! This section was re-written.

File: CONTRIBUTING.md (hunk 1 above)
Comment (diverdane) [reply CONTRIBUTING.md:170]: Yes. This was re-written.

File: docker-compose.yml (hunk 2 above)
Comment (diverdane) [reply docker-compose.yml:5]: I'm seeing Postgres version `10.14` still being used in Conjur, e.g.:
https://github.com/cyberark/conjur/blob/master/dev/docker-compose.yml#L4
Should this be `10.15` everywhere?

Comment (diverdane) [reply docker-compose.yml]: I changed this to version `10.15` to match what's used in cyberark/conjur-oss-helm-chart.

**Review by BradleyBoutcher — APPROVED** [review]
LGTM

**Pull request merged**